src/streams/EditStreamCard.js

```diff
@@ -13,33 +24,55 @@ const componentStyles = StyleSheet.create({
   },
 });
 
+type SelectorProps = {|
+  isAdmin: boolean,
+|};
+
 type Props = $ReadOnly<{|
+  dispatch: Dispatch,
   isNewStream: boolean,
   initialValues: {
     name: string,
     description: string,
     invite_only: boolean,
+    stream_post_policy: number,
   },
-  onComplete: (name: string, description: string, isPrivate: boolean) => void,
+  onComplete: (
+    name: string,
+    description: string,
+    isPrivate: boolean,
+    streamPostPolicy: number,
+  ) => void,
+
+  ...SelectorProps,
 |}>;
 
 type State = {|
   name: string,
   description: string,
   isPrivate: boolean,
+  streamPostPolicyIndex: number,
 |};
 
-export default class EditStreamCard extends PureComponent<Props, State> {
-  state = {
-    name: this.props.initialValues.name,
-    description: this.props.initialValues.description,
-    isPrivate: this.props.initialValues.invite_only,
-  };
+class EditStreamCard extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      name: this.props.initialValues.name,
+      description: this.props.initialValues.description,
+      isPrivate: this.props.initialValues.invite_only,
+      streamPostPolicyIndex:
+        STREAM_POST_POLICIES.findIndex(
+          policy => policy.key === this.props.initialValues.stream_post_policy,
+        ) || 0,
```

Comment:
This alternation between "index" and "key" seems pretty fragile -- a sharp edge in the API of `SelectionList.Selector`.

Can you revise it so that instead of `selectedIndex` it takes `selectedKey`? And then everything can work with keys and forget about indexes, except the `renderItem` callback in `OptionsScreen`.

Reply:
Initially I was passing `selectedKey`, actually it was making `Selector` complicated. As now it has to lookup in whole list to find label for the selected item. 
So I thought going with `selectedIndex`.

Reply:
I'm not sure what you mean by "label" -- that isn't a term used in this code, except for something quite different.

There shouldn't be a need to go look through the whole list to find something. If you're referring to `onOptionSelect` -- it can just pass the key to the callback.

Reply:
By `label` I mean text which is show to user for each option.

So, I am referring to the currently selected option text shown in the selector.

Let's define a data structure like this:
```js
STREAM_POST_POLICIES = {
    StreamPostPolicies.everyone: {label: "All stream members"},
    .........
}
```

and pass this to `SelectionList`. And it can take `selectedKey`.

Not in `Selector` it will be easy to get selected option text (label), 
```
options[selectedKey].label
```

now but in `OptionsScreen ` we have to do `Objects.keys` to convert it into array

Reply:
Changed the data structure of the selection list options, now it is
```
export type OptionsType = $ReadOnly<{| [key: string]: { label: string } |}>;
```

and `selectedKey: string,` is passed everywhere.